api/src/main/java/org/apache/iceberg/types/Comparators.java

```diff
@@ -149,6 +155,51 @@ public int compare(List<T> o1, List<T> o2) {
     }
   }
 
+  private static class MapComparator<K, V> implements Comparator<Map<K, V>> {
+    private final Comparator<K> keyComparator;
+    private final Comparator<V> valueComparator;
+    private final Comparator<List<K>> keyListComparator;
+
+    private MapComparator(Types.MapType mapType) {
+      this.keyComparator = internal(mapType.keyType());
```

Comment:
Shouldn't `internal` function also have a branch for the map type comparator?

Reply:
As @nastra pointed out maybe a few more test cases might worth the effort, I'd add some more for nested types.